generate.go

```diff
@@ -85,6 +85,19 @@ type LinkMap map[string]*Link
 // CfgMap is meant to represent a map with values of one or more unknown types
 type CfgMap map[string]interface{}
 
+func Join(cfgs ...*CfgMap) (CfgMap, error) {
+	c := CfgMap{}
+	for _, cfg := range cfgs {
+		for k, v := range *cfg {
+			if _, ok := c[k]; ok {
+				return nil, fmt.Errorf("Duplicate key found in two contexts: %s", k)
```

Comment:
My first question that came up in my head upon first seeing this PR was how to handle collisions. I like your solution of returning an error since it is clear and unambiguous. To make sure I understand, that means that the order of the `<ctx>...` arguments would not matter, right? (like `cogs gen examples/2.http.cog.toml get post` vs. ``cogs gen examples/2.http.cog.toml post get`)

Reply:
exactly, I'm a bit leery of introducing a waterfall merge for things like `PORT`